src/client/pages/Login/ChangePassword/ChangePassword.tsx

```diff
@@ -0,0 +1,69 @@
+import './ChangePassword.scss'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useNavigate, useParams } from 'react-router'
+
+import { ApiEndPoint } from 'meta/api/endpoint'
+
+import { useGetRequest } from 'client/hooks/getRequest'
+import Form from 'client/components/Form'
+
+import { useFormDefinition } from './hooks/useFormDefinition'
+import { useOnSuccess } from './hooks/useOnSuccess'
+import { useValidationSchema } from './hooks/useValidationSchema'
+
+const ChangePassword: React.FC = () => {
+  const navigate = useNavigate()
+  const { t } = useTranslation()
+  const { resetPasswordUuid } = useParams<{ resetPasswordUuid: string }>()
+
+  const [email, setEmail] = useState<string>('')
+
+  const { data, dispatch: fetchData } = useGetRequest(ApiEndPoint.User.resetPassword(), {
+    params: { resetPasswordUuid },
+  })
+
+  useEffect(() => {
+    if (resetPasswordUuid) fetchData()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [resetPasswordUuid])
+
+  useEffect(() => {
+    if (data?.user?.email) {
+      setEmail(data.user.email)
+    }
+  }, [data])
```

Comment:
1. hooks better in separate files
2. it might be better to show skeletons until the data hasn't been fetched. you would also avoid this unnecessary [email,setEmail].
it would be:
```
  return (
    <div className="change-password">
    { data && <ChangePasswordForm  data={data} /> }
    { !data && <Skeletons... }
    </div>
```

what do you think ?

Reply:
can you please post a video @sorja instead ? difficult to see how the skeleton works